src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs

```diff
@@ -228,6 +228,21 @@ public IEnumerable<ZipFilePair> EnumeratePackageEntries(IEnumerable<string> pack
             }
         }
 
+        /// <summary>
+        /// Validate all files in package are not traversed outside of the expected extraction path.
+        /// </summary>
+        public async Task<bool> ValidatePackageEntries(string destination, CancellationToken token)
```

Comment:
Do you need a destination path to be passed here?

Can't you use a dummy value?

Reply:
rename to ValidatePackageEntriesAsync

Reply:
on client side ,we don't need this API, we always have real value, it's only for server.

Reply:
You don't need a destination though do you?

What happens if you harcode `C:\.nuget\packages` for example?

The value of the destination does not change/impact the result. 
It should not be a parameter.